app/models/mixins/supports_feature_mixin.rb

```diff
@@ -226,35 +119,45 @@ def unsupported
 
   class_methods do
     # This is the DSL used a class level to define what is supported
-    def supports(feature, &block)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
-      define_supports_feature_methods(feature, &block)
+    def supports(feature, description: nil, &block)
+      define_supports_feature_methods(feature, :description => description, &block)
     end
 
     # supports_not does not take a block, because its never supported
     # and not conditionally supported
-    def supports_not(feature, reason: nil)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
-      define_supports_feature_methods(feature, :is_supported => false, :reason => reason)
+    def supports_not(feature, description: nil, reason: nil)
+      define_supports_feature_methods(feature, :is_supported => false, :description => description, :reason => reason)
     end
 
     # query the class if the feature is supported or not
     def supports?(feature)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
+      guard_queryable_feature(feature)
       public_send("supports_#{feature}?")
     end
 
     # query the class for the reason why something is unsupported
     def unsupported_reason(feature)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
+      guard_queryable_feature(feature)
       feature = feature.to_sym
       public_send("supports_#{feature}?") unless unsupported.key?(feature)
       unsupported[feature]
     end
 
     # query the class if a feature is generally known
     def feature_known?(feature)
-      SupportsFeatureMixin::QUERYABLE_FEATURES.key?(feature.to_sym)
+      queryable_features.key?(feature.to_sym)
+    end
+
+    def guard_queryable_feature(feature)
```

Comment:
This is the one method I've never understood...  just about every guard comes before a public_send call, which will blow up with a NoMethodError anyway, so it just blows up with a different error.  If we have proper tests it will blow up in either case, so it's redundant.

The other guard is around looking up the unsupported reason, and those I think make more sense.

Reply:
This is actually causing the failures https://travis-ci.com/github/ManageIQ/manageiq/jobs/482423680#L768-L778 because `queryable_features` for the subclass doesn't include the `queryable_features` from the parent classes.

If we just let the methods inherit normally instead of checking for hash keys this would all work

Reply:
To be honest, I do not understand this method either.  This PR was focused on leaving the logic mostly untouched except for removing the global constant QUERYABLE_FEATURES that needs to be updated every time we add a feature anywhere.  So, I would be for removing this and simplifying the code even further, if we can come to agreement.

Reply:
Yeah we're going to have to do something because this guard is keeping child classes from inheriting their parent's supported methods (since the queryable_features hash isn't inherited)

```
>> ExtManagementSystem.supports?(:refresh_ems)
=> true
>> ManageIQ::Providers::BaseManager.supports?(:refresh_ems)
Traceback (most recent call last):
       14: from bin/rails:4:in `<main>'
       13: from bin/rails:4:in `require'
       12: from railties (6.0.3.5) lib/rails/commands.rb:18:in `<top (required)>'
       11: from railties (6.0.3.5) lib/rails/command.rb:46:in `invoke'
       10: from railties (6.0.3.5) lib/rails/command/base.rb:69:in `perform'
        9: from thor (1.1.0) lib/thor.rb:392:in `dispatch'
        8: from thor (1.1.0) lib/thor/invocation.rb:127:in `invoke_command'
        7: from thor (1.1.0) lib/thor/command.rb:27:in `run'
        6: from railties (6.0.3.5) lib/rails/commands/console/console_command.rb:102:in `perform'
        5: from railties (6.0.3.5) lib/rails/commands/console/console_command.rb:19:in `start'
        4: from railties (6.0.3.5) lib/rails/commands/console/console_command.rb:70:in `start'
        3: from (irb):5
        2: from app/models/mixins/supports_feature_mixin.rb:134:in `supports?'
        1: from app/models/mixins/supports_feature_mixin.rb:153:in `guard_queryable_feature'
SupportsFeatureMixin::UnknownFeatureError (Feature ':refresh_ems' is unknown to SupportsFeatureMixin.)
```

I'd be :+1: for removing it all together

Reply:
Deleted!

Reply:
Fixed:
```
>> ManageIQ::Providers::BaseManager.supports?(:refresh_ems)
=> true
>> ExtManagementSystem.supports?(:refresh_ems)
=> true
```